packages/axelar-wasm-std/src/utils.rs

```diff
@@ -19,6 +22,22 @@ impl<T> TryMapExt<T> for Vec<T> {
     }
 }
 
+/// since the `Felt` type doesn't error on overflow, we have to implement that check
+pub fn does_felt_overflow_from_slice(felt_hex_slice: &[u8]) -> bool {
+    if felt_hex_slice.len() > 32 {
+        return true;
+    }
+    let felt_max_hex_str = format!("{:064x}", Felt::MAX);
+    U256::from_be_slice(felt_hex_slice) > U256::from_be_hex(&felt_max_hex_str)
+}
+
+/// since the `Felt` type doesn't error on overflow, we have to implement that check
+pub fn does_felt_overflow_from_str(felt_hex_str: &str) -> bool {
```

Comment:
can't we just check if the first nibble (4 bits) is non-zero?

Reply:
I thought that as well. 
It's a 252 bit number, but not 252 bit MAX number, 2^252-1 (252 one values).
The first bit is one, but the other ones are not all ones. This is what it looks like, represented as 256 bits:

`0000 1000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0001 0001 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000`

So checking the first 4 bits for non-zero is not enough.